### DISCUSSION TITLE
Paths and environment variables in artifact paths

### DISCUSSION THREAD

**Discussion by pmatos** [community/community#25576] [Actions]
<p>Hello,</p>
<p>I am having some issue finding the path to where a checkout happens. I could of course, set an environment variable to the <code>$PWD</code> in a step but then I can’t use that in an artifact path since as far as I understand an artifact path does not expand shell variables.</p>
<p>If I save a file in a step, how can I get that path to the artifact path so that it is saved?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>There is a default environment variable named <strong>GITHUB_WORKSPACE.</strong></p>
<p><em>The GitHub workspace directory path. The workspace directory contains a subdirectory with a copy of your repository if your workflow uses the <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Factions%2Fcheckout&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C47d916055cf045e99d5008d76cc9e383%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637097486667666317&amp;sdata=5g7PY9PPnsWbMJ96rqodrc2rkBcQpjHJypoYqXcUp%2Fc%3D&amp;reserved=0" rel="nofollow noopener">actions/checkout</a> action. If you don’t use the actions/checkout action, the directory will be empty. For example, /home/runner/work/my-repo-name/my-repo-name.</em></p>
<p>If you don’t set working-directory keyword in your step, the default working directory is /home/runner/work/my-repo-name/my-repo-name.</p>
<p>Let me show you an example, there is my file structure in the repo:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a9b0f77c88ed41843d326236dfbec20b1aec134f.png" data-download-href="/uploads/short-url/od9QcFFUHGVmCo450nJS6Pq9oLl.png?dl=1" title="repo dir.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181091872-8f5b1bcd-4df3-4c32-91ba-fda795d89e02.png" alt="repo dir.png" data-base62-sha1="od9QcFFUHGVmCo450nJS6Pq9oLl" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/a9b0f77c88ed41843d326236dfbec20b1aec134f_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/2X/a/a9b0f77c88ed41843d326236dfbec20b1aec134f_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/a/a9b0f77c88ed41843d326236dfbec20b1aec134f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a9b0f77c88ed41843d326236dfbec20b1aec134f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">repo dir.png</span><span class="informations">1075×564 36.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p> After checkout action, these folders and files in the root of my repo will exist under /home/runner/work/orgrepo1/orgrepo1</p>
<p>I want to upload contents under projects/frontend folder. Then I use <strong>path: ./projects/frontend/</strong> in actions/upload-artifact@v1.0.0.</p>
<p>name: publish artifacts</p>
<p>on: push</p>
<p>jobs:</p>
<p>  build:</p>
<p>    runs-on: ubuntu-latest</p>
<p>    steps:</p>
<p>      - uses: actions/checkout@v1</p>
<p>      - name: Upload artifact</p>
<p>        uses: actions/upload-artifact@v1.0.0</p>
<p>        with:</p>
<p>          # Artifact name</p>
<p>          name: drop1</p>
<p>          # Directory containing files to upload</p>
<p>          path: ./projects/frontend/</p>
